docs/pages/build/submitting-to-app-stores.md

```diff
@@ -0,0 +1,106 @@
+
+**Important:**
+
+- You have to create a [Google Service Account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) and download its JSON private key.
+- After that, you'll have to create an app on [Google Play Console](https://play.google.com/apps/publish/) and upload your app manually at least once.
+
+#### Creating a Google Service Account
+
+See [expo.fyi/creating-google-service-account](https://expo.fyi/creating-google-service-account) to learn more.
+
+#### Manually uploading your app for the first time
+
+Before using `eas submit -p android` for uploading your builds, you have to upload your app manually at least once. This is a limitation of the Google Play Store API.
+
+See [expo.fyi/first-android-submission](https://expo.fyi/first-android-submission) to learn more.
+
+#### Using EAS CLI to upload the further builds of your app
+
+To upload your Android app to the Google Play Store, run `eas submit --platform android` and follow the instructions on the screen. The command will perform the following steps:
+
+1. Log in to your Expo account and ensure that your app project exists on EAS servers
+2. Prompt for the Android package name unless `expo.android.package` is set in app configuration or the `--android-package` param is provided.
+3. Ask for a binary to submit. You can select one of the following:
+
+   - The latest finished Android build for the project on EAS servers
+   - Specific build ID. The ID can be found on your EAS Account under _Builds v2_ section
+   - Path to an APK or AAB archive on your local filesystem
+   - URL to the app archive
+
+   > This step can be skipped if one of the following parameters is provided: `--latest`, `--id`, `--path` or `--url`.
+
+4. When you chose local path or URL as your binary location, the command will try to autodetect archive type basing on file extension. If it's unable to do so, it may prompt you to specify if it's APK or AAB archive. You can also use a `--type` parameter.
+5. Unless a `--key` param is provided, you will be prompted for path to your Google Services JSON key.
+6. The summary of provided configuration is displayed and the submission process begins. The submission progress is displayed on the screen.
+7. Your build should now be visible on Google Play Console. If something goes wrong, an appropriate message is displayed on the screen.
+
+### 2.2 Submitting your iOS app to App Store
+
+**Important**: You'll have to generate an [Apple app-specific-password](https://expo.fyi/apple-app-specific-password) before uploading an app using EAS CLI.
+
+To upload your iOS app to the Apple App Store, run `eas submit --platform ios` and follow the instructions on the screen. The command will perform the following steps:
+
+1. Log in to Expo account and ensure that your app project exists on EAS servers
+2. Ensure that your app exists on App Store Connect and its [Bundle Identifier](https://expo.fyi/bundle-identifier) is registered on Apple Dev Center:
```

Comment:
```suggestion
2. Ensure that your app exists on App Store Connect and its [Bundle Identifier](https://expo.fyi/bundle-identifier) is registered on Apple Developer Portal:
```
?

Reply:
> its [Bundle Identifier](https://expo.fyi/bundle-identifier) is registered on Apple Dev Center:

Technically, if the bundle identifier hasn't been registered on Apple Developer Portal it's not possible to have an app binary using that particular bundle id. So maybe the command should fail/or ask the user why they provided a non-existent bundle id?

Reply:
Makes sense, but what about situation, when user builds a binary with specific bundle ID, then revokes it manually on Developer Portal and after that tries to upload id to the store? I know this sounds irrational, but if bundle ID is required for ASC Submission, it should be checked right before to avoid unnecessary failures.